pkg/driver/node.go

```diff
@@ -848,29 +839,104 @@ func collectMountOptions(fsType string, mntFlags []string) []string {
 	return options
 }
 
+// startNotReadyTaintWatcher launches a short‑lived Node informer that removes the
+// ebs.csi.aws.com/agent‑not‑ready taint as soon as both the taint is present and
+// the driver’s Allocatable value is set. The watcher stops automatically after
+// maxWatchDuration or immediately after successful removal.
+func startNotReadyTaintWatcher(clientset kubernetes.Interface, maxWatchDuration time.Duration) {
+	nodeName := os.Getenv("CSI_NODE_NAME")
+	if nodeName == "" {
+		klog.V(4).InfoS("CSI_NODE_NAME missing, skipping taint watcher")
+		return
+	}
+
+	stopCh := make(chan struct{})
+	var closeOnce sync.Once
+	closeWatch := func() { closeOnce.Do(func() { close(stopCh) }) }
+
+	factory := informers.NewSharedInformerFactoryWithOptions(
+		clientset,
+		0,
+		informers.WithTweakListOptions(func(lo *metav1.ListOptions) {
+			lo.FieldSelector = "metadata.name=" + nodeName
+		}),
+	)
+	informer := factory.Core().V1().Nodes().Informer()
+
+	attemptTaintRemoval := func(n *corev1.Node) {
+		if !hasNotReadyTaint(n) {
+			return
+		}
+		if err := checkAllocatable(clientset, nodeName); err != nil {
+			return
+		}
+		if err := removeNotReadyTaint(clientset); err != nil {
+			klog.ErrorS(err, "failed to remove agent‑not‑ready taint")
+			return
+		}
+		klog.V(2).InfoS("Successfully removed agent‑not‑ready taint; stopping watcher")
+		closeWatch()
```

Comment:
Why are we stopping here? Isn't the point of this bugfix that the taint gets re-added and thus needs to be re-removed?

Reply:
It's not that the taint gets re-added, its more that there are no taints to remove when the check is performed, so the taint removal is skipped, see logs in the original bug report:

```
...
I0418 20:03:42.343483       1 node.go:896] "No taints to remove on node, skipping taint removal"
```

I think not stopping the informer for the full `taintWatcherDuration` makes sense though to handle the scenario you mentioned (a controller re-applying the taint after it is removed), especially given its short lifespan.